### ISSUE TITLE
Sequential docker run via nginx gives 500

### ISSUE THREAD

**Issue by jonbjo** [travis-ci/travis-ci#5386]
TL;DR;
The second sequential docker run that calls an API in Tomcat via NGINX fails with status 500.

I have 4 containers involved in my Travis build
* leanbean-proxy which is NGINX
* leanbean-api which is Tomcat
* leanbean-db which is MySQL
* leanbean-dev which has all the build tools installed

In the before_script step I run
```bash
docker run leanbean-build sh -c 'cd /opt/leanbean/leanbean-api/ && bash deploy-tomcat.sh test'
```
and the deploy-tomcat.sh runs
```bash
mvn --batch-mode -Dmaven.tomcat.url=http://leanbean-proxy/manager/text tomcat7:redeploy
```
This gives a status 200 back and I can also see the PUT in logs from leanbean-proxy and the deployment of the war file in logs from leanbean-api.

Later in the script step I run
```bash
docker run leanbean-build sh -c 'cd /opt/leanbean/ && bash test.sh leanbean-proxy'
```
and the test.sh runs
```bash
curl -H "Content-Type: application/json" -s -o /dev/null -w "%{http_code}" -X POST -d '{"alias":"Bob","uuid":"1ab9f8ta452bg2eq2"}' http://leanbean-proxy/leanbean/v1/devices/
```
This gives a status of 500 back and I can't see any traces of the request in Tomcat logs, only in NGINX.

I cannot replicate this behaviour when running it manually. Both the deployment and the test scripts runs successfully and returns 200. Since I can't reproduce it outside of Travis build environment, I am thinking that I am setting up the containers wrong or something else is going wrong on that end. I have a vague memory of reading of someone having a similar problem when running the same container many times in a sequence on Travis CI, but I can't find that issue/thread again. All help is appreciated because I don't know how to debug this further. 

Source code can be found at https://github.com/wisobi/leanbean and build history at https://travis-ci.org/wisobi/leanbean/builds/

**Comment by BanzaiMan** [comment]
Email us at support@travis-ci.com and we'll get you set up with a debug VM, so you can investigate interactively.

Thanks.

**Comment by joepvd** [comment]
Closing.
